comms/lighthouse/src/peerjs-server/instance.ts

```diff
@@ -40,24 +40,24 @@ export const createInstance = ({
     config
   })
 
-  wss.on('connection', (client: IClient) => {
+  wss.on('connection', async (client: IClient) => {
```

Comment:
careful, wss will ignore the Promise returned by this function. Failing silently or killing the program, or leading to leaks.

Reply:
Nice catch. I'll see what's the best way to handle this. I'm thinking since WSS already has an `error callback, maybe I should try-catch and emit `error`?

Reply:
I have many ideas, I don't know what are the implications of the 'error' event. For instance, in web sockets, whenever you receive an 'error' event, you can assume the socket is closed. 

The one I like the most is creating a HighOrderFunction that forces us to consider corner cases, like this:

```ts
function asyncCallback(fn: (args) => Promise<any>): (args) => void {
  return (...args) => 
    fn.apply(null, args).catch(err => {
      console.log('error in async callback, the program will finish')
      console.error(err)
      process.exit(1)
    })
}

  wss.on('connection', asyncCallback(async (client: IClient) => {
    try {
    ..
    } catch(e) {
      // explicitly handle every error for each case
    }
  })
```

The idea is to force us to program for corner cases, because the program will exit if we don't

Reply:
Hmmm, but I'm not really sure we want to be so forceful. This code (peerjs-server) has been running for a while with no issues, and I wouldn't want to make it crash on every error that we don't explicitly handle.

It is true that we are making modifications of peerjs server, and that could introduce new issues that maybe we should handle. But I'd rather wait until modifications are more significant before introducing a change in behaviour so significant. I prefer to limit the changes if possible.

Reply:
Very well, please be mindful to properly handle the "catch" at least